### PR TITLE
Stanford template override stuff

### PR DESCRIPTION
**Pull request by willgearty**
This is a work in progress of integrating lots of Stanford template override changes into main, since a number of them are useful for other sites. Some of these are fixes, some are additions, and some might not even be useful, but I tried to include whatever I thought my be good for main. I've still got a couple big template overrides, but this covers all of the smaller things.

- [x] ~~teachereventsmodule/event_signup.html~~ (implemented in #2709)
- [x] profiles/...
- [x] users/student_schedule.html
- [x] users/teacherbio.html
- [x] users/userview_class_entry.html
- [x] ~~users/loginbox_content.html~~ (part of theme)
- [x] ~~teacherclassregmodule/classedit.html~~ (highly customized)
- [x] ~~teacherclassregmodule/listclasses.html~~ (highly customized)
- [x] ~~volunteersignup/signup.html~~ (changes are being implemented in #2583)

### CODE REVIEW

**Comment by willgearty** [comment]
Here are some screenshots of the changes to the teacher events manage and teach pages. Essentially, I've made two big changes:

1. Instead of super boring dropdown menus for the trainings and interviews, I've converted the options into large, pretty buttons via jquery which the user can select from to change the dropdown selections (also via jquery).
2. Allow for descriptions to have newline characters (\n) to specify when there should be newlines within the buttons.

If it would be nice, I also have some jquery that processes the newline characters for the manage page display, but I thought it might be useful (for main) to just keep it with the exact description.

image

image

**Comment by willgearty** [comment]
Here's what the "Classes Taught" section of userview looks like:

image

**Comment by willgearty** [comment]
The `float: none` additions may only be applicable to the Stanford theme, since part of our theme wants to push some of those buttons to the right via `float: right`, but I figured it would be good to just make sure they aren't getting moved in other themes.
Another thing I noticed (e.g. on the teacherbio page), is that the circles theme uses a blue color for visited links, so any time an `<a>` tag is used with the `btn-primary` class, it's hard to see the text in the button.

Otherwise, happy to provide examples of anything else!

File: esp/templates/program/modules/studentregphasezero/status.html [hunk 1]
```diff
@@ -92,7 +92,7 @@ <h2>
 <form action="#" method="POST">
 <!--<b>Specify Role Name for Winners: </b>--><input type="hidden" name="rolename" value="{{role}}"></br></br>
 <input type="checkbox" name="confirm">  I confirm that I would like to run the student lottery.</br></br>
-<input type="submit" value="Run Student Lottery" />
+<input type="submit" style="float:none" value="Run Student Lottery" />
```
[the remark this answers is not in the document]
Comment (willgearty) [reply esp/templates/program/modules/studentregphasezero/status.html]: It's possible this only matters for the Stanford theme, but we have some theme stuff that adds `float: right` to `input`s, so I just put this here to make sure these buttons are nice and centered and not floating.

File: esp/templates/program/modules/adminclass/manageclass.html [hunk 2]
```diff
@@ -172,7 +172,7 @@ <h1>Manage Class {{ class.id }}: {{ class.title }}</h1>
             None requested
         {% else %}
         {% for rr in form.sec.getResourceRequests %}
-            {{ rr.res_type.name }} <br />
+            {{ rr.res_type.name }}: {{ rr.desired_value }}<br />
```
[the remark this answers is not in the document]
Comment (willgearty) [reply esp/templates/program/modules/adminclass/manageclass.html]: Ah, looks like the default `desired_value` is blank, so I'll add an if here.

[the remark this answers is not in the document]
Comment (willgearty) [reply esp/templates/program/modules/adminclass/manageclass.html]: Actually, the default might be "Don't care", now that I think about it. Do we want to display that? Or show nothing if that's the `desired_value`

File: esp/templates/program/modules/adminclass/manageclass_class_info.html [hunk 3]
```diff
@@ -32,6 +32,12 @@
         {{ class.prereqs }}
     </td>
 </tr>
+<tr>
```
[the remark this answers is not in the document]
Comment (willgearty) [reply esp/templates/program/modules/adminclass/manageclass_class_info.html]: Oh, I guess that would be good if it's optional.

File: esp/templates/program/modules/teachereventsmodule/event_signup.html [hunk 4]
```diff
@@ -2,9 +2,157 @@
```
[the remark this answers is not in the document]
Comment (willgearty) [reply esp/templates/program/modules/teachereventsmodule/event_signup.html]: Check out the preview in my comment on the PR.

[the remark this answers is not in the document]
Comment (willgearty) [reply esp/templates/program/modules/teachereventsmodule/event_signup.html]: And also, probably merits a separate PR.

[the remark this answers is not in the document]
Comment (willgearty) [reply esp/templates/program/modules/teachereventsmodule/event_signup.html]: Moved to #2709

**Comment by willgearty** [comment]
This pull request **introduces 2 alerts** when merging 9d8e75e1f4c2837e05ff9b6deee94abb4e4083ef into 1e4fa5bbcc67e85097105c61aeb1e0179f502f7c - [view on LGTM.com](https://lgtm.com/projects/g/learning-unlimited/ESP-Website/rev/pr-193d8d64387c8393397e3a4c53900fb16fd89df7)

**new alerts:**

* 2 for Potentially unsafe external link

---

*Comment posted by [LGTM.com](https://lgtm.com)*

File: esp/templates/users/teacherbio.html [hunk 5]
```diff
@@ -56,9 +59,7 @@ <h2>Brief Biographical Sketch:</h2>
   <br />
   <br />
   <h2>Past Classes</h2> &nbsp; 
-  <small>(Look at the <a href="/archives/classes/" title="Past ESP Classes">class archive</a> for more.)</small>
```
[the remark this answers is not in the document]
Comment (willgearty) [reply esp/templates/users/teacherbio.html]: Given that this is a biography for a specific teacher, I'm not really sure this is the place to link to the class archive anyways.

**Comment by willgearty** [comment]
This should be good for a final review @benjaminjkraft !

**Comment by willgearty** [comment]
This pull request **introduces 2 alerts** when merging 0884825f48a641d557889ecf5b17a8a2b8efdc0d into 1e4fa5bbcc67e85097105c61aeb1e0179f502f7c - [view on LGTM.com](https://lgtm.com/projects/g/learning-unlimited/ESP-Website/rev/pr-50b227132390b71029bf1c61617f312622055962)

**new alerts:**

* 2 for Potentially unsafe external link

---

*Comment posted by [LGTM.com](https://lgtm.com)*

File: esp/templates/users/userview.html [hunk 6]
```diff
@@ -30,6 +30,9 @@ <h1>User Information</h1>
             {% endfor %}
         </select>
     </form>
+    {% if profile.teacher_info %}
```
Comment (kkbrum) [inline esp/templates/users/userview.html]: This doesn't catch my account, which is a teacher & student & admin. It catches people who are just teachers, but if we want to include this we should probably make it catch the case of teacher being just one of the roles?

Comment (kkbrum) [reply esp/templates/users/userview.html]: (same with line 97 though)

Comment (willgearty) [reply esp/templates/users/userview.html]: Hmm...this is probably a broader issue related to how the profile works, but I should be able to fix this by just doing something like `{% if "Teacher" in user.getUserTypes %}`.

Comment (willgearty) [reply esp/templates/users/userview.html]: (And then same for all the other user type conditionals)

**Pull request merged**